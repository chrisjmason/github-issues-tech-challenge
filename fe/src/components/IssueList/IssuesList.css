.IssueList {
  display: block;
  border: 1px solid green;
}

.IssueListItem {
  margin: 10px 0;
  display: block;
  background-color: lightgray;
}
