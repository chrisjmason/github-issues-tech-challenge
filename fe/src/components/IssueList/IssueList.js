import React from "react";
import "./IssuesList.css";

const IssueList = (props) => {
  const { issues } = props;
  return (
    <ul className="IssueList" data-testid="issue-list">
      {issues.map((issue) => (
        <li key={issue.id} className="IssueListItem">
          <div>{issue.title}</div>
          <div>{issue.description}</div>
          <span>{issue.issueId}</span>
        </li>
      ))}
    </ul>
  );
};

export default IssueList;
