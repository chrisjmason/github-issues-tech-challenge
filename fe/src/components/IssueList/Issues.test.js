import React from "react";
import { render, screen } from "@testing-library/react";
import IssueList from "./IssueList";

describe("Layout", () => {
  const mockData = [
    {
      id: 1,
      title: "Bug: React App with router not opening on ios 13.4.x",
      issueId: "#18699",
      description: "opened 5 hours ago by nvnvenki",
    },
    {
      id: 2,
      title: "Bug: anther bugs",
      issueId: "#18620",
      description: "opened 5 hours ago by nvnvenki",
    },
  ];
  it("matches snapshot", () => {
    render(<IssueList issues={mockData} />);
    expect(screen.getByTestId("issue-list")).toMatchSnapshot();
  });
});
