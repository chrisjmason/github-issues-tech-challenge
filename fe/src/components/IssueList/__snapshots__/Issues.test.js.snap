// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<ul
  class="IssueList"
  data-testid="issue-list"
>
  <li
    class="IssueListItem"
  >
    <div>
      Bug: React App with router not opening on ios 13.4.x
    </div>
    <div>
      opened 5 hours ago by nvnvenki
    </div>
    <span>
      #18699
    </span>
  </li>
  <li
    class="IssueListItem"
  >
    <div>
      Bug: anther bugs
    </div>
    <div>
      opened 5 hours ago by nvnvenki
    </div>
    <span>
      #18620
    </span>
  </li>
</ul>
`;
